platform/security/handler/security-handler-basic/src/main/java/org/codice/ddf/security/handler/basic/WssBasicAuthenticationHandler.java

```diff
@@ -38,42 +36,63 @@
+    }
+
+    public WssBasicAuthenticationHandler(Parser parser) {
+        this.parser = parser;
+    }
+
     protected BaseAuthenticationToken getBaseAuthenticationToken(String realm, String username,
             String password) {
         UsernameTokenType usernameTokenType = new UsernameTokenType();
         AttributedString user = new AttributedString();
         user.setValue(username);
         usernameTokenType.setUsername(user);
+        String usernameToken = null;
 
         // Add a password
         PasswordString pass = new PasswordString();
         pass.setValue(password);
         pass.setType(WSConstants.PASSWORD_TEXT);
         JAXBElement<PasswordString> passwordType = new JAXBElement<>(QNameConstants.PASSWORD,
-                PasswordString.class, pass);
-        usernameTokenType.getAny().add(passwordType);
+                PasswordString.class,
+                pass);
+        usernameTokenType.getAny()
+                .add(passwordType);
+
         // Marshall the received JAXB object into a DOM Element
-        String usernameToken = null;
-        Writer writer = new StringWriter();
-        try {
-            Set<Class<?>> classes = new HashSet<>();
-            classes.add(ObjectFactory.class);
-            classes.add(
-                    org.apache.cxf.ws.security.sts.provider.model.wstrust14.ObjectFactory.class);
+        List<String> ctxPath = new ArrayList<>(2);
+        ctxPath.add(ObjectFactory.class.getPackage()
+                .getName());
+        ctxPath.add(org.apache.cxf.ws.security.sts.provider.model.wstrust14.ObjectFactory.class.getPackage()
+                .getName());
+        if (null != parser) {
+            ParserConfigurator configurator = parser.configureParser(ctxPath,
+                    WssBasicAuthenticationHandler.class.getClassLoader());
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
 
-            JAXBContextCache.CachedContextAndSchemas cache = JAXBContextCache
-                    .getCachedContextAndSchemas(classes, null, null, null, false);
-            JAXBContext jaxbContext = cache.getContext();
+            JAXBElement<UsernameTokenType> tokenType =
+                    new JAXBElement<>(QNameConstants.USERNAME_TOKEN,
+                            UsernameTokenType.class,
+                            usernameTokenType);
 
-            Marshaller marshaller = jaxbContext.createMarshaller();
-            JAXBElement<UsernameTokenType> tokenType = new JAXBElement<>(
-                    QNameConstants.USERNAME_TOKEN, UsernameTokenType.class, usernameTokenType);
-            marshaller.marshal(tokenType, writer);
-            usernameToken = writer.toString();
-        } catch (JAXBException ex) {
-            LOGGER.warn("", ex);
+            try {
+                parser.marshal(configurator, tokenType, os);
+                usernameToken = os.toString();
+            } catch (ParserException ex) {
+                LOGGER.warn("", ex);
```

Comment:
It'd be great to add a comment in the logger about what was unable to be parsed instead of an empty string